### DISCUSSION TITLE
WebQuery POST doesn't seem to do anything

### DISCUSSION THREAD

**Discussion by GijsTimmers** [arendst/Tasmota#15799] [Q&A] [answered]
I'm trying to set up a [Pushover](https://pushover.net) notification on my Tasmota 11.1.0 ESP8285 device.

Pushover is set up correctly, as issuing the following command from a Linux box works:

```
curl \
  --form-string "token=application_token" \
  --form-string "user=user_token" \
  --form-string "message=hello world" \
  https://api.pushover.net/1/messages.json
```

However, the following command on Tasmota doesn't work:

`WebQuery https://api.pushover.net/1/messages.json POST {"token":"application_token", "user":"user_token", "message": "hello world"}`

It merely shows that Tasmota executed the command:

```
12:54:16.090 CMD: WebQuery https://api.pushover.net/1/messages.json POST {"token":"application_token", "user":"user_token", "message": "hello world"}
12:54:16.092 SRC: WebConsole from 192.168.1.230
12:54:16.095 CMD: Grp 0, Cmd 'WEBQUERY', Idx 1, Len 149, Pld -99, Data 'https://api.pushover.net/1/messages.json POST {"token":"application_token", "user":"user_token", "message": "hello world"}'
12:54:16.176 RSL: RESULT = {"WebQuery":"Done"}
```

I compiled Tasmota with this in user_config_override.h hoping that I would get extra output:

```
#ifndef USE_WEBSEND_RESPONSE
#define USE_WEBSEND_RESPONSE
#endif
```

but it shows the same response.

All help is appreciated.

**Comment by sfromis** [comment, marked as answer]
Encrypted `https` does not work on ESP8266, only on ESP32. Use `http` if you can, or ....

**Reply by GijsTimmers** [reply]
Thank you, I switched to [ntfy.sh](https://ntfy.sh/), which works without HTTPS. All working now!

**Reply by pkkrusty** [reply]
Can you share your ntfy.sh post url? Not the topic, obviously, just want to see the syntax.

**Reply by GijsTimmers** [reply]
`WebQuery http://ntfy.sh/topic POST [Title: Dishwasher] Program finished.`

**Reply by pkkrusty** [reply]
Thanks.  For posterity, the webquery will fail if the underscore is used i.e. `http://ntfy.sh/topic_three POST [Title: Dishwasher] Program finished.` will result in `MQT: water-heater/stat/RESULT = {"WebQuery":"Connect failed"}`

**Comment by marrold** [comment]
For anyone stumbling across this in the future, at the time of writing Pushover will accept a POST over HTTP _without_ TLS, but you must force the character encoding header like so:

    WebQuery http://api.pushover.net/1/messages.json POST [Content-Type:application/json] {"token":"SECRET_TOKEN", "user":"SECRET_TOKEN", "message": "IT WORKS"}
